hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +50,123 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
+        ArrayHandle<unsigned int> h_embed_group(m_embed_group->getIndexArray(),
+                                                access_location::host,
+                                                access_mode::read);
+        ArrayHandle<Scalar4> h_vel_embed(m_pdata->getVelocities(),
+                                         access_location::host,
+                                         access_mode::read);
+        ArrayHandle<unsigned int> h_body_embed(m_pdata->getBodies(),
+                                               access_location::host,
+                                               access_mode::read);
+        ArrayHandle<unsigned int> h_rtag_embed(m_pdata->getRTags(),
+                                               access_location::host,
+                                               access_mode::read);
+
+        // check if some of the masses are less or equal to 0
+        bool invalid_mass = false;
+        bool central_interacting = false;
+        for (unsigned int idx = 0; idx < N_tot && !invalid_mass && !central_interacting; ++idx)
+            {
+            // get the index from the embedded group
+            unsigned int particle_index = h_embed_group.data[idx];
+            // check mass
+            const Scalar4 vel_mass = h_vel_embed.data[particle_index];
+            const Scalar mass = vel_mass.w;
+            if (mass <= Scalar(0))
+                {
+                invalid_mass = true;
+                }
+            // check if particle is central particle in rigid body
+            unsigned int central_tag = h_body_embed.data[particle_index];
+            assert(central_tag <= m_pdata->getMaximumTag());
+            if (central_tag < MIN_FLOPPY)
+                {
+                unsigned int central_idx = h_rtag_embed.data[central_tag];
+                if (particle_index == central_idx)
+                    {
+                    central_interacting = true;
+                    }
+                }
+            }
+
+#ifdef ENABLE_MPI
+// MPI_Reduce invalid_mass using logical or if communicating
+#endif // ENABLE_MPI
+
+        if (invalid_mass)
+            {
+            m_exec_conf->msg->warning() << "Some particles have a mass <= 0, may lead to "
+                                           "invalid results during momentum transfer."
+                                        << std::endl;
+            }
+        if (central_interacting)
+            {
+            m_exec_conf->msg->warning() << "Central particle of rigid body included in "
+                                           "MPCD collision. Check if this is intentional."
+                                        << std::endl;
+            }
+        }
+    m_checked_collision_warnings = true;
     }
 
+void mpcd::CollisionMethod::beginRigidBodyCollision(uint64_t timestep)
+    {
+    if (m_embed_group)
+        {
+        unsigned int N_tot = m_embed_group->getNumMembers();
+        m_initial_velocity.resize(N_tot);
```

Comment:
This resize should only be done if necessary using a swap idiom:
```suggestion
        if (N_tot > m_initial_velocity.getNumElements())
            {
            GPUArray<Scalar4> initial_velocity(N_tot, m_exec_conf);
            m_initial_velocity.swap(initial_velocity);
            }
```
I would actually also move this up into the `collide` method, right before you call `beginRigidBodyCollision`, so that way it doesn't need to be duplicated in the GPU version of this method.